File: Difficulty: Basic/Last Coin in a Game of Alternates/last-coin-in-a-game-of-alternates.cpp
class Solution {
  public:
    int coin(vector<int>& arr) {
        // code here
        int left = 0;
        int right = arr.size() - 1;
    
        // Remove one coin in each move
        while (left < right) {
    
            // Greedily remove larger end coin
            if (arr[left] >= arr[right]) {
                left++;
            } else {
                right--;
            }
        }
    
        // Last remaining coin
        return arr[left];
    }
};

File: Difficulty: Basic/Last Coin in a Game of Alternates/last-coin-in-a-game-of-alternates.java
class Solution {
    int min (int a, int b) {
        return a < b ? a : b;
    }
    public int coin(int[] arr) {
        /*It is just a find the smallest problem*/
        int m = Integer.MAX_VALUE;
        for (int x: arr) m = min(m, x);
        return m;
    }
}